WhisperAPI/WhisperAPI/Services/SuggestionsService.cs

```diff
@@ -20,17 +20,33 @@ public SuggestionsService(IIndexSearch indexSearch, INlpCall nlpCall, List<strin
             this._irrelevantsIntents = irrelevantsIntents;
         }
 
-        public IEnumerable<SuggestedDocument> GetSuggestions(string querry)
+        public IEnumerable<SuggestedDocument> GetSuggestions(List<SearchQuerry> querriesList)
         {
-            var nlpAnalysis = this._nlpCall.GetNlpAnalysis(querry);
+            IEnumerable<SuggestedDocument> suggestedDocuments = new List<SuggestedDocument>();
+            var allMessages = string.Join(" ", querriesList.Select(m => m.Querry));
+            var nlpAnalysis = this._nlpCall.GetNlpAnalysis(querriesList.LastOrDefault()?.Querry);
 
             // TODO: filter out the querry from lq when the irrelevantsIntents matches the ones in the settings and when persistance will be done
             if (this.IsIntentRelevant(nlpAnalysis))
             {
-                return this.SearchCoveoIndex(querry);
+                var coveoIndexDocuments = this.SearchCoveoIndex(allMessages);
+                suggestedDocuments = this.FilterOutChoosenSuggestions(coveoIndexDocuments, querriesList);
             }
 
-            return new List<SuggestedDocument>();
+            // TODO: Send 5 most relevant results
+            return suggestedDocuments.Take(5);
+        }
+
+        private IEnumerable<SuggestedDocument> FilterOutChoosenSuggestions(
```

Comment:
This method could be public since it is stateless so we can Unit test it (although don't add it to the interface, keep it in the class).

This method is perfect, but we will keep an eye on it as it may not scale well eventually.